pkg/codegen/internal/test/testdata/types.json

```diff
@@ -421,6 +484,43 @@
         "requiredNumber",
         "requiredString"
       ]
+    },
+    "typetests:regressions:pulumi/pulumi/7454": {
+      "description": "Regression test for pulumi/pulumi#7454",
+      "properties": {
+        "nestedArray": {
+          "type": "array",
+          "items": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          },
+          "language": {
+            "test": {
+              "expected": {
+                "dotnet": {
+                  "input": "InputList<ImmutableArray<string>>",
```

Comment:
I'm having to do some memorization here, is this table driven?

Assuming `InputList<X>` ~ `Input<Seq<X>>`, this is `Input<Seq<Seq<string>>>`. Why is this not `Input<Seq<Input<Seq<Input<str>>>>>` as in the other languages?

 
A(S)

Reply:
> Why is this not Input<Seq<Input<Seq<Input<str>>>>> as in python + nodejs?

That's a great question--and one I don't have an answer for. My original changes to reify input types did change the definition to what you'd expect, but that turns out to be a breaking change because (among other things) this no longer compiles:

```csharp
InputList<ImmutableArray<string>> = foo.someNestedArrayInputProperty;
```